docs/web3.main.rst

```diff
@@ -49,6 +49,85 @@ Attributes
        'Geth/v1.4.11-stable-fed692f6/darwin/go1.7'
 
 
+.. _batch_requests:
+
+Batch Requests
+~~~~~~~~~~~~~~
+
+.. py:method:: Web3.batch_requests()
+
+    The JSON-RPC API allows for batch requests, meaning you can send a single request
+    that contains an array of RPC commands. Generally, this may be useful when you want
+    to limit the number of requests you send to a node.
+
+    You have can choose to build a batch of requests within or outside of a context manager:
+
+    .. code-block:: python
+
+        with w3.batch_requests() as batch:
+            batch.add(w3.eth.get_block(6))
+            batch.add(w3.eth.get_block(4))
+            batch.add(w3.eth.get_block(2))
+
+            responses = batch.execute()
+            assert len(responses) == 3
+
+    Using the batch object directly:
+
+    .. code-block:: python
+
+        batch = w3.batch_requests()
+        batch.add(w3.eth.get_block(1))
+        batch.add(w3.eth.get_block(2))
+        responses = batch.execute()
+        assert len(responses) == 2
+
+    Contract interactions can be included in batch requests by omitting the ``call`` method:
```

Comment:
Unless you have the contract call documentation in front of you, I feel like this *omitting* can be a bit confusing. Maybe this helps a tiny bit?

```suggestion
    Contract interactions can be included in batch requests by omitting the ``call()`` method:
```